tools/json2json.cpp

```diff
@@ -72,33 +72,18 @@ int main(int argc, char *argv[]) {
     std::cerr << "warning: ignoring everything after " << argv[optind + 1]
               << std::endl;
   }
-  auto [p, error] = simdjson::padded_string::load(filename);
-  if (error) {
-    std::cerr << "Could not load the file " << filename << std::endl;
-    return EXIT_FAILURE;
-  }
-  simdjson::ParsedJson pj;
-  int res =
-      simdjson::json_parse(p, pj); // do the parsing, return false on error
-  if (res != simdjson::SUCCESS) {
-    std::cerr << " Parsing failed. Error is '" << simdjson::error_message(res)
+  simdjson::document::parser parser;
+  auto [doc, error] = parser.load(filename); // do the parsing, return false on error
+  if (error != simdjson::SUCCESS) {
+    std::cerr << " Parsing failed. Error is '" << simdjson::error_message(error)
               << "'." << std::endl;
     return EXIT_FAILURE;
   }
   if (apidump) {
-    simdjson::ParsedJson::Iterator pjh(pj.doc);
-    if (!pjh.is_ok()) {
-      std::cerr << " Could not iterate parsed result. " << std::endl;
-      return EXIT_FAILURE;
-    }
-    compute_dump(pjh);
+    simdjson::document::iterator i(doc);
```

Comment:
Can we replace this with `cout << doc` or is compute_dump doing something different than that?

Reply:
Trying to discourage the use of document::iterator and ParsedJson::Iterator both now that we have the DOM API. Actually, I think we need to remove document::iterator.

Reply:
Point taken. I removed this whole thing entirely. The program is now much leaner... which is the point... our own tools should be good citizens.

benchmark/bench_dom_api.cpp has references to document::iterator though maybe it is intentional.